R/PseudoBulk.R

```diff
@@ -1231,3 +1194,80 @@ FitRegularizedClassificationGlm <- function(seuratObj,
     ))
   }
 }
+
+#' @title .PrepareDataForClassificationGlm
+#' @description a helper function for FitClassificationGlm to remove special characters from the gene names.
+#' @param seuratObj Input seurat object used for training a glmnet model. 
+#' @param assay Seurat Object's assay
+#' @param layer layer within the Seurat object assay. Recommended to be "scale.data".
+#' @param rescale The feature selection will optimize for "heatmap-interpretable genes" so the features are intended to be scaled. If TRUE, this will rescale the variable features.
+#' @param numberOfVariableFeatures A parameter to select how many features should be selected as variable for scaling, by default, all genes will be used.
+#' @param excludeVariableGenes A boolean option to exclude variable genes from the analysis. Only the Variable Gene Exclusion list (v2) from RIRA is supported. 
+#' @param metadataVariableForClassification The metadata feature to be classified. If non-binary, then multinomial regression will automatically be performed.
+#' 
+#' @return a matrix with the special characters removed from the gene names.
+
+.PrepareDataForClassificationGlm <- function(seuratObj, 
+                                             assay = "RNA", 
+                                             layer = 'scale.data', 
+                                             rescale = TRUE, 
+                                             numberOfVariableFeatures = NULL,  
+                                             excludeVariableGenes = T, 
+                                             metadataVariableForClassification = NULL) {
+  # rescale the input data (in case of an upstream subset since it was last rescaled).
+  
+  if (excludeVariableGenes) {
+    exclusionList <- NULL
+  } else {
+    exclusionList <- RIRA::GetGeneSet("VariableGenes_Exclusion.2")
```

Comment:
why not push that into an argument, with a default value of RIRA::GetGeneSet("VariableGenes_Exclusion.2")?

Reply:
and even better idea is to make the default value as the string "VariableGenes_Exclusion.2", and then call the RIRA function that expands gene names using RIRA's gene sets. let me know if this isnt enough to find that function.